Add `ignoreMediaFeatureNameValues: {}` to `media-feature-name-value-no-unknown` (#8976)
stylelint/stylelint: lib/rules/media-feature-name-value-no-unknown/README.md

```css
@media (width: 10px) { top: 1px; }
```

### `ignoreMediaFeatureNameValues`

```json
{
  "ignoreMediaFeatureNameValues": {
    "media-feature-name": ["array", "of", "values", "/regex/"]
  }
}
```

Ignore the specified media feature name and value pairs. Keys in the object indicate media feature names.

`media-feature-name` must be a valid media feature name.

Given:

```json
{
  "media-feature-name-value-no-unknown": [
    true,
    {
      "ignoreMediaFeatureNameValues": {
        "display-mode": ["/^my-/", "custom"]
      }
    }
  ]
}
```

The following patterns are _not_ considered problems:

<!-- prettier-ignore -->
```css
@media (display-mode: my-feature) {}
```

<!-- prettier-ignore -->
```css

> @media (display-mode: custom) {}

Reviewer: [suggestion] What if respecting the examples above for the `true` section?

```suggestion
        "color": ["/^my-/", "custom"]
      }
    }
  ]
}
```

The following patterns are _not_ considered problems:

<!-- prettier-ignore -->
```css
@media (color: my-feature) {}
```

<!-- prettier-ignore -->
```css
@media (color: custom) {}
```

```css
@media (width: 10px) { top: 1px; }
```

### Optional secondary options

#### `ignoreMediaFeatureNameValues`

```json
{
  "ignoreMediaFeatureNameValues": {
    "media-feature-name": ["array", "of", "values", "/regex/"]
  }
}
```

Ignore the specified media feature name and value pairs. Keys in the object indicate media feature names.

> `media-feature-name` must be a valid media feature name.

Reviewer 2: ```suggestion
You can specify a regex for a media feature name, such as { "/^color/": [] }.
```

Given:

```json
{
  "media-feature-name-value-no-unknown": [
    true,
    {
      "ignoreMediaFeatureNameValues": {

>         "color": ["/^my-/", "custom"]

Reviewer 2: ```suggestion
        "/^color/": ["--foo", "/^--bar/"]
```
